### DISCUSSION TITLE
v0.10.0-beta1

### DISCUSSION THREAD

**Discussion by laipz8200** [langgenius/dify#8948] [Releases]
## ✨ What’s New in v0.10.0-beta1?✨

> [!IMPORTANT] 
> This version is for preview only, do not use in production!

> [!NOTE]
> We highly value your feedback! Please feel free to share your thoughts and discuss new features using the GitHub Discussions feature. You can find the entry point for Discussions at the bottom of the release page.

We are pleased to offer Dify 0.10.0-beta1 for early access. This release includes support for file processing in Workflows and Chatflow, allowing users to utilize `File` and `Array<File>` variable types and pass these variables through multiple nodes.

### 🚀 Key Features

- **File Processing in Workflows**: Added support for `File` and `Array<File>` variable types, allowing seamless handling and transfer of files across various nodes in a workflow. This capability significantly expands the range of tasks you can automate with Dify.

- **Document Extractor Node**: Introducing the new Document Extractor node, which serves as a case example for file usage. This node is designed to parse various types of documents into text, facilitating text extraction from different file formats for further processing.

- **List Filter Node**: To make it easier for users to work with `Array` variables, the List Filter node has been added. This node allows you to filter, sort, and slice arrays based on specific conditions, offering enhanced control over array-type data within your workflows.

### 🎨 UI and UX Updates

We have also introduced some UI and UX improvements to make Dify more modern and user-friendly.

| ![Screenshot 1](https://github.com/user-attachments/assets/a1328b8a-2437-492b-8b8c-50e38a1d93ae) | ![Screenshot 2](https://github.com/user-attachments/assets/6184572b-8a76-4d9f-ad9b-60eb279b7784) | ![Screenshot 3](https://github.com/user-attachments/assets/318db481-c02d-48f9-972e-446eaba7f9fc) |
|---|---|---|

### Quick Start

To upgrade to this version, follow these steps:

1. Clone the repository using the following command:

   ```bash
   git clone https://github.com/langgenius/dify.git
   ```

2. Switch to the beta branch:

   ```bash
   git checkout 0.10.0-beta1
   ```

3. Then follow the [Deploy with Docker Compose > Starting Dify](https://docs.dify.ai/getting-started/install-self-hosted/docker-compose#starting-dify).

We hope you find the new file processing capabilities and nodes useful in this release. Your feedback and contributions are always appreciated to help us improve Dify!

<hr /><em>This discussion was created from the release <a href='https://github.com/langgenius/dify/releases/tag/0.10.0-beta1'>v0.10.0-beta1</a>.</em>

**Comment by timonqqq** [comment]
hi,
I just upgraded Dify to version 0.10.0-beta1, but I'm encountering the following error (as shown in the image). Does anyone have any ideas on how to diagnose this?
![image](https://github.com/user-attachments/assets/986abe06-80de-41c3-a8d9-5aa7c051ce59)

**Reply by timonqqq** [reply]
By the way, I had no issues with version 0.9.1 using the same method and configuration. I upgraded specifically to try out the new file handling features.

**Reply by laipz8200** [reply]
Thank you for the report. We have identified some bugs related to the SSRF proxy configuration, which cause errors when starting with Docker Compose. These will be fixed in the next version.

In the meantime, you can try starting the project from the source code.

**Comment by zzbzc** [comment]
请问一下，云服务该如何更新呢？？
May I ask, how should cloud services be updated??

**Reply by laipz8200** [reply]
The Cloud version will be updated after the community version is released and its stability has been tested. Please stay tuned.
